query/security/ldap/src/main/java/org/codice/ddf/admin/ldap/commons/ServerGuesser.java

```diff
@@ -126,23 +131,45 @@ public static ServerGuesser buildGuesser(Connection connection) {
   }
 
   public List<String> getGroupObjectClass() {
-    return ImmutableList.of("groupOfNames", "group", "posixGroup");
+    return new ArrayList<>(
+        getResults("(|(objectClass=group)(objectClass=group*)(objectClass=*Group*))")
+            .stream()
+            .flatMap(entry -> Entries.getObjectClasses(entry).stream())
+            .filter(GROUP)
+            .map(ObjectClass::getNameOrOID)
```

Comment:
✏️ Instead of calling `getNameOrOID` twice - once explicitly here and once implicitly in the Predicate for the `filter` above, can switch the order of the stream operations and update the `GROUP` Predicate to 
`name -> name.toLowerCase().contains("group")`